save-demo/src/main/kotlin/com/saveourtool/save/demo/utils/HttpUtils.kt

```diff
@@ -90,3 +94,42 @@ suspend fun <R> demoAgentRequestWrapper(
             }
         }
 }
+
+/**
+ * Request wrapper that uses existing kubernetes service [DEMO_SUBDOMAIN_NAME] created for all save-demo-agents,
+ *   ip address of pod (Pod.status.podIp) and sends [request] to constructed url with path [urlPathSegments].
+ *
+ * Url is constructed like this:
+ *
+ * {POD-IP}.{NAMESPACE}.svc.{CLUSTER.DOMAIN}
+ *
+ * Note that [request] returns object of generic type [R] allowing to either return HttpResponse or body of the response.
+ *
+ * @param urlPathSegments list of endpoint path segments (e.g. ["run"])
+ * @param demoPod [Pod] with save-demo-agent that runs requested demo
+ * @param request callback that receives [Url]
+ * @return result of type [R]
+ */
+suspend fun <R> demoAgentRequestWrapper(
+    urlPathSegments: List<String>,
+    demoPod: Pod,
+    request: suspend (Url) -> R,
+): R {
+    val host = addressToDnsResolution(demoPod.status.podIP)
+    return request(
+        URLBuilder(
+            host = host,
+            port = SAVE_DEMO_AGENT_DEFAULT_PORT,
+            pathSegments = urlPathSegments,
+        ).build()
+    )
+}
+
+private fun addressToDnsResolution(podIp: String, namespace: String = "save-cloud") = listOf(
```

Comment:
suggestion to extract to a public function and add tests

Reply:
Ok, will do it